xarray/tests/test_dataarray.py

```diff
@@ -3647,7 +3647,9 @@ def test_to_and_from_dict_with_time_dim(self) -> None:
         t = pd.date_range("20130101", periods=10)
         lat = [77.7, 83.2, 76]
         da = DataArray(x, {"t": t, "lat": lat}, dims=["t", "lat"])
-        roundtripped = DataArray.from_dict(da.to_dict())
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", message="Converting non-nanosecond")
+            roundtripped = DataArray.from_dict(da.to_dict())
```

Comment:
if we have this pattern in multiple modules, it might be worth adding the code as a special context manager to `xarray.tests.__init__`. Something like this might work (I didn't check):
```python
from contextlib import contextmanager
import warnings

@contextmanager
def ignore_warnings(category=None, pattern=None):
    if category is None and pattern is None:
        raise ValueError("need at least one of category and pattern")

    try:
        with warnings.catch_warnings():
            warnings.filterwarnings("ignore", message=pattern, category=category)
            yield
    finally:
        pass
```

Reply:
Thanks—I ended up switching to marking these tests with `@pytest.mark.filterwarnings("ignore:Converting non-nanosecond")`, since that is a pattern we use elsewhere in the tests already.